HPCC-25319 Support using cert-manager to generate PKI certificates

afishbeck opened the pull request:
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

afishbeckln committed:
HPCC-25319 Support using cert-manager to generate PKI certificates

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexisrisk.com>

afishbeck commented:
I'll look at the windows build issues later.

richardkchapman commented:
@afishbeck Needs rebasing

mckellyln commented on helm/examples/certmanager/README.md:
```diff
@@ -0,0 +1,200 @@
+
+## Jetstack cert-manager support:
+
+The following wil use cert-manager to automatically provision and manage TLS certificates for the
```
will

mckellyln reviewed (commented):
Looks good.
I will change my calls to be:
createSecureSocketContextSecret()
after this is merged

richardkchapman commented:
@afishbeck Is this still a draft?

HPCCSmoketest commented:
Process of PR-14587, label: certManager6 starts now.
The reason of this test is: Forced to re-test.
Commit ID: b9f365b22cdeadf9725555f83094574ffc32601d
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-7.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-7.ca-central-1.compute.internal
Sha: b9f365b22cdeadf9725555f83094574ffc32601d
Number of Error(s): 5
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:734:26: error: \xe2\x80\x98EC_KEY_priv2buf\xe2\x80\x99 was not declared in this scope
checking for pthread_setname_gnu (by compile).../home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:738:13: error: \xe2\x80\x98OPENSSL_clear_free\xe2\x80\x99 was not declared in this scope
make[2]: *** [system/jlib/CMakeFiles/jlib.dir/build.make:479: system/jlib/CMakeFiles/jlib.dir/jsecrets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1302: system/jlib/CMakeFiles/jlib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-03-15_13-17-09
time logs started
Prep time   :     8 sec (00:00:08)
Build time  :   334 sec (00:05:34)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   342 sec (00:05:42)
time logs end

[Build error:](https://gist.github.com/8bf39f46805b4f47a668d8d709bf72f5)
Build log [Build log](https://gist.github.com/HPCCSmoketest/8bf39f46805b4f47a668d8d709bf72f5#file-result-21-03-15-13-11-15-log)

HPCCSmoketest commented:
Process of PR-14587, label: certManager6 starts now.
The reason of this test is: Forced to re-test.
Commit ID: fbed8233c5a2e0b53246c3bb2f1b14e058d2d93b
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-76.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-76.ca-central-1.compute.internal
Sha: fbed8233c5a2e0b53246c3bb2f1b14e058d2d93b
Number of Error(s): 5
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:734:26: error: \xe2\x80\x98EC_KEY_priv2buf\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:738:13: error: \xe2\x80\x98OPENSSL_clear_free\xe2\x80\x99 was not declared in this scope
make[2]: *** [system/jlib/CMakeFiles/jlib.dir/build.make:479: system/jlib/CMakeFiles/jlib.dir/jsecrets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1302: system/jlib/CMakeFiles/jlib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-03-15_15-00-17
time logs started
Prep time   :     8 sec (00:00:08)
Build time  :   338 sec (00:05:38)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   346 sec (00:05:46)
time logs end

[Build error:](https://gist.github.com/f4a9a7a0e5a642446000c80a24abd553)
Build log [Build log](https://gist.github.com/HPCCSmoketest/f4a9a7a0e5a642446000c80a24abd553#file-result-21-03-15-14-54-18-log)

ghalliday reviewed (commented):
@afishbeck looks good - walk through is really good.  A few minor comments.  I wasn't really sure why internal was the right name, but I suspect I haven't properly understood it.

afishbeck commented on helm/examples/certmanager/README-vault-pki.md:
````diff
@@ -0,0 +1,343 @@
+ecl run --ssl hthor examples/certmanager/localhttpcall.ecl
+```
+
+Note that for the HTTPCALL in our ecl example the url now starts with "local:" this tells HTTPCALL/SOAPCALL to use the local client certificate, and to verify the server using the local certificate authority certificate.
````
Yes.  It's for using mtls within the local environment.  Perhaps mtls: would be better, I'll change it to that unless we think of something better.  It boils down to "use the client and CA certs from the local TLS secret".

afishbeck replied:
I thought about trying to detect that the url was local.. but didn't want to outsmart myself and end up regretting it.

afishbeck commented on helm/hpcc/templates/_helpers.tpl:
```diff
@@ -886,3 +890,179 @@ Pass in dict with root, pod, target and type
+  commonName: {{ $name }}.{{ $namespace }}
+  isCA: false
+  privateKey:
+    algorithm: ECDSA
```
Yes I just did what I could to get the key size right.  The cert itself isn't intended for roxie to use.  At some point I can play with it or we can come up with a better way of generating those keys.  I just thought this could be a way to do it for now.  When there is a vault available it may have other ways to generate the keys (but would belong in a different PR for sure).

afishbeck replied:
So just to clarify, I thought we needed a max key size of 256 and the certificate rules are:
```
	// Size is the key bit size of the corresponding private key for this certificate.
	// If `algorithm` is set to `RSA`, valid values are `2048`, `4096` or `8192`,
	// and will default to `2048` if not specified.
	// If `algorithm` is set to `ECDSA`, valid values are `256`, `384` or `521`,
	// and will default to `256` if not specified.
```
Since only the actual key is being used I can match the algorithm to the desired key size.

afishbeck replied:
Also I think Roxie is using AES for UPD encryption.

"AES is a symmetric algorithm which uses the same 128, 192, or 256 bit key for both encryption and decryption (the security of an AES system increases exponentially with key length). "

afishbeck commented:
@ghalliday regarding the term "internal".  I changed several places to use local rather than internal.  I'll consider changing the remaining places.

afishbeck commented on system/jlib/jsecrets.cpp:
```diff
@@ -713,3 +717,72 @@ extern jlib_decl bool getSecretValue(StringBuffer & result, const char *category
+    if (eckey)
+    {
+        unsigned char *priv = NULL;
+        size_t privlen = EC_KEY_priv2buf(eckey, &priv);
```
It was implicitly including via a 3rd party header, but I will make it explicit above.

richardkchapman commented on roxie/ccd/ccdmain.cpp:
```diff
@@ -772,6 +773,13 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
         soapTraceLevel = topology->getPropInt("@soapTraceLevel", runOnce ? 0 : 1);
         miscDebugTraceLevel = topology->getPropInt("@miscDebugTraceLevel", 0);
 
+//for draft tesing only, remove before PR
```
Should this be removed now?

afishbeck replied:
yes, will do.

richardkchapman reviewed (changes requested):
A couple of minor questions

afishbeck commented on esp/applications/eclservices/esp.yaml:
```diff
@@ -14,14 +14,13 @@ esp:
+     privatekey: /opt/HPCCSystems/secrets/certificates/internal/tls.key
+     cipherList: "ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5"
+     verify:
+       enable: true
```
This flag basically turns on mutual TLS.  Since this is an internal service, if TLS is on it should be mutual TLS.  On the other hand we don't want to require client certificates for public services.

afishbeck replied:
I'll add a comment since it can be confusing.

HPCCSmoketest commented:
Process of PR-14587, label: certManager6 starts now.
The reason of this test is: Forced to re-test.
Commit ID: aafbbc571a2a753a05f74898a29d10c61ef72e36
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-189.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-189.ca-central-1.compute.internal
Sha: aafbbc571a2a753a05f74898a29d10c61ef72e36
Conflicting files, should skip build and test.

[Build error:](https://gist.github.com/eb1b95258ea4ed174609c89d850b3697)
Build log [Build log](https://gist.github.com/HPCCSmoketest/eb1b95258ea4ed174609c89d850b3697#file-result-21-03-21-00-49-17-log)

HPCCSmoketest commented:
Process of PR-14587, label: certManager6 starts now.
The reason of this test is: Base branch changed.
Commit ID: 77d143359442b09b3ac1d607e30d06473903d613
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-240.ca-central-1.compute.internal

afishbeck commented:
@ghalliday @richardkchapman 
Updated:
1. Changed terms "external" and "internal",  to "public" and "local".  The old terms are over used and less clear.
2. Changed SOAPCALL prefix for using client certificates from "local" to "mtls" to make it clearer what is happening.
3. Cleaned up template code.
4. Tested and improved the ability to add an externally created certificate.  For example someone getting an official cert for eclwatch.

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-240.ca-central-1.compute.internal
Sha: 77d143359442b09b3ac1d607e30d06473903d613
Number of Error(s): 5
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:737:26: error: \xe2\x80\x98EC_KEY_priv2buf\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:741:13: error: \xe2\x80\x98OPENSSL_clear_free\xe2\x80\x99 was not declared in this scope
checking if gcc supports -std=gnu89 flag... make[2]: *** [system/jlib/CMakeFiles/jlib.dir/build.make:479: system/jlib/CMakeFiles/jlib.dir/jsecrets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1302: system/jlib/CMakeFiles/jlib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-03-22_15-24-10
time logs started
Prep time   :     8 sec (00:00:08)
Build time  :   331 sec (00:05:31)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   339 sec (00:05:39)
time logs end

[Build error:](https://gist.github.com/5eca2442c3c1b50e3cf1449b415b100c)
Build log [Build log](https://gist.github.com/HPCCSmoketest/5eca2442c3c1b50e3cf1449b415b100c#file-result-21-03-22-15-18-18-log)

HPCCSmoketest commented:
Process of PR-14587, label: certManager6 starts now.
The reason of this test is: Forced to re-test.
Commit ID: c0bb8909c95448b1684f129e866c6db8d60060f0
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-206.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-206.ca-central-1.compute.internal
Sha: c0bb8909c95448b1684f129e866c6db8d60060f0
Number of Error(s): 5
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:734:26: error: \xe2\x80\x98EC_KEY_priv2buf\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:738:13: error: \xe2\x80\x98OPENSSL_clear_free\xe2\x80\x99 was not declared in this scope
make[2]: *** [system/jlib/CMakeFiles/jlib.dir/build.make:479: system/jlib/CMakeFiles/jlib.dir/jsecrets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1302: system/jlib/CMakeFiles/jlib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-03-22_17-18-08
time logs started
Prep time   :     9 sec (00:00:09)
Build time  :   329 sec (00:05:29)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   338 sec (00:05:38)
time logs end

[Build error:](https://gist.github.com/bd7f42df3340b26addb53d11d22e4c63)
Build log [Build log](https://gist.github.com/HPCCSmoketest/bd7f42df3340b26addb53d11d22e4c63#file-result-21-03-22-17-12-18-log)

HPCCSmoketest commented:
Process of PR-14587, label: certManager6 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 54baf7b0b4c8b43bd4b02c2ff9bf9d4235580443
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-235.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-235.ca-central-1.compute.internal
Sha: 54baf7b0b4c8b43bd4b02c2ff9bf9d4235580443
Number of Error(s): 5
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:738:26: error: \xe2\x80\x98EC_KEY_priv2buf\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:742:13: error: \xe2\x80\x98OPENSSL_clear_free\xe2\x80\x99 was not declared in this scope
checking whether we are cross compiling... make[2]: *** [system/jlib/CMakeFiles/jlib.dir/build.make:479: system/jlib/CMakeFiles/jlib.dir/jsecrets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1302: system/jlib/CMakeFiles/jlib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-03-22_18-37-13
time logs started
Prep time   :     9 sec (00:00:09)
Build time  :   336 sec (00:05:36)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   345 sec (00:05:45)
time logs end

[Build error:](https://gist.github.com/bfbda9d0b9d334fa11f2c20421268cc2)
Build log [Build log](https://gist.github.com/HPCCSmoketest/bfbda9d0b9d334fa11f2c20421268cc2#file-result-21-03-22-18-31-16-log)

afishbeck commented:
@richardkchapman squashed and rebased.

HPCCSmoketest commented:
Process of PR-14587, label: certManager6 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 5cdbe2c08d46314c8c00d13b65f2ac2cacb9ef14
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-252.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-252.ca-central-1.compute.internal
Sha: 5cdbe2c08d46314c8c00d13b65f2ac2cacb9ef14
Number of Error(s): 5
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:736:26: error: \xe2\x80\x98EC_KEY_priv2buf\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14587/HPCC-Platform/system/jlib/jsecrets.cpp:740:13: error: \xe2\x80\x98OPENSSL_clear_free\xe2\x80\x99 was not declared in this scope
make[2]: *** [system/jlib/CMakeFiles/jlib.dir/build.make:479: system/jlib/CMakeFiles/jlib.dir/jsecrets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1302: system/jlib/CMakeFiles/jlib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-03-26_18-50-13
time logs started
Prep time   :     8 sec (00:00:08)
Build time  :   336 sec (00:05:36)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   344 sec (00:05:44)
time logs end

[Build error:](https://gist.github.com/ac37679bb3d361400845035d5ad27e55)
Build log [Build log](https://gist.github.com/HPCCSmoketest/ac37679bb3d361400845035d5ad27e55#file-result-21-03-26-18-44-15-log)

HPCCSmoketest commented:
Process of PR-14587, label: certManager6 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 306ad22bba14dc4dc9d29b02b929c0fb3d44a4b6
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-47.ca-central-1.compute.internal

afishbeck commented:
@AttilaVamos not sure about "cp: cannot stat '/home/runner/work/HPCC-Platform/HPCC-Platform/src/.github/workflows/smoketest-preabort.sh': No such file or directory"

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-47.ca-central-1.compute.internal
Sha: 306ad22bba14dc4dc9d29b02b929c0fb3d44a4b6
Build: success
Milestone:Install hpccsystems-platform-community_8.0.1-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |150 |150 |0 |0 |0 |57 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |25 sec  (00:00:25) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |19 sec  (00:00:19) |
|test (hthor) |959 |959 |0 |791 sec  (00:13:11) |
|test (thor) |871 |871 |0 |1076 sec  (00:17:56) |
|test (roxie) |1032 |1032 |0 |895 sec  (00:14:55) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  603 sec (00:10:03) |  106 sec (00:01:46) |   21 sec (00:00:21) |   18 sec (00:00:18) | 1389 sec (00:23:09) |   19 sec (00:00:19) | 2165 sec (00:36:05) |

ghalliday commented on helm/hpcc/values.yaml:
```diff
@@ -115,6 +115,53 @@ storage:
+## The default configuration is meant to provide reasonable functionality without additional dependencies.
+##
+## Public issuers can be tricky if you want browsers to recognize the certificates.  This is a complex topic outside the scope
+## of this comment.  The default for the public issuer generates self signed certifiecates. The expectation is that this will be
```
typo: certificates

ghalliday reviewed (approved):
one trivcial typo
